zephyr/lib/vpages.c

```diff
@@ -0,0 +1,340 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2025 Intel Corporation.
+ *
+ * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <sof/lib/regions_mm.h>
+#include <zephyr/drivers/mm/mm_drv_intel_adsp_mtl_tlb.h>
+
+LOG_MODULE_REGISTER(vpage, CONFIG_SOF_LOG_LEVEL);
+
+/* Simple Page Allocator.
+ *
+ * This allocator manages the allocation and deallocation of virtual memory pages from
+ * a predefined virtual memory region which is larger than the physical memory region.
+ *
+ * Both memory regions are divided into 4kB pages that are represented as blocks in a
```

Comment:
maybe better not hard-code page size - throughout the code. `CONFIG_MM_DRV_PAGE_SIZE`